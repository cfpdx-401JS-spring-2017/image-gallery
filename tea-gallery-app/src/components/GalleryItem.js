import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DeleteImageButton } from './DeleteButtons';

class GalleryItem extends Component {

  static PropTypes = {
    images: PropTypes.array.required,
    onDeleteImage: PropTypes.func.required
  };

  constructor(props) {
    super(props);

    this.state = {
      index: 0
    };
  }

  changeImage(increment) {
    const images = this.props.images;
    let index = this.state.index;
    const image = images[index];

    index += increment;
    if (index === images.length) index = 0;
    if (index === -1 || isNaN(index)) index = images.length - 1;

    this.setState({ index: index });
    this.setState({ images: image });
  }

  render() {
    const { images } = this.props;
    return (
      <div className="gallery-img-wrapper">
        {images.length > 0 &&
          <div>
            <img className="gallery-img" src={images[this.state.index].url} alt="tea"></img>
            <button className="back-button" onClick={() => this.changeImage(-1)}>Back</button>
            <button className="next-button" onClick={() => this.changeImage(+1)}>Next</button>
            <DeleteImageButton
              images={images}
              onDeleteImage={() => this.props.onDeleteImage(images[this.state.index]._id)}
            />
          </div>
        }
        {images.length < 1 && <h2>Album is empty. Add images below</h2>}
      </div>
    );
  }
}

export default GalleryItem;