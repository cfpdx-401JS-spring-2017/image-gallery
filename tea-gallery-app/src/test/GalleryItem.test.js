import React from 'react';
import renderer from 'react-test-renderer';
import GalleryItem from '../components/GalleryItem';

const images = [
  {
    title: 'tea leaves in teapot',
    description: 'what nice leaves',
    url: '/../public/img/teainpot.jpg'
  },
  {
    title: 'Tetsubin over charcoal',
    description: 'what a nice tetsubin',
    url: '/../public/img/tetsubin-charcoal.jpg'
  },
  {
    title: 'Teapot display',
    description: 'what a nice teapot display',
    url: '/../public/img/teapot-display.jpg'
  }
];

describe('GalleryItem', () => {
  it('renders the GalleryItem', () => {
    const component = renderer.create(<GalleryItem images={images}/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});