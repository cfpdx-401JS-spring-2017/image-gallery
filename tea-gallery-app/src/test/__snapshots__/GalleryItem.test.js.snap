// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem renders the GalleryItem 1`] = `
<div
  className="gallery-img-wrapper"
>
  <div>
    <img
      alt="tea"
      className="gallery-img"
      src="/../public/img/teainpot.jpg"
    />
    <button
      className="back-button"
      onClick={[Function]}
    >
      Back
    </button>
    <button
      className="next-button"
      onClick={[Function]}
    >
      Next
    </button>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
